rest/diagnostic_api.go

```diff
@@ -0,0 +1,188 @@
+//  Copyright 2013-Present Couchbase, Inc.
+//
+//  Use of this software is governed by the Business Source License included
+//  in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+//  in that file, in accordance with the Business Source License, use of this
+//  software will be governed by the Apache License, Version 2.0, included in
+//  the file licenses/APL2.txt.
+
+package rest
+
+import (
+	"github.com/couchbase/sync_gateway/auth"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/gorilla/mux"
+)
+
+// each field in this response is a map of collection names to a map of channel names to grant history info
+type getAllChannelsResponse struct {
+	AdminGrants                          map[string]map[string]auth.GrantHistory            `json:"admin_grants,omitempty"`
+	DynamicGrants                        map[string]map[string]auth.GrantHistory            `json:"dynamic_grants,omitempty"`
+	DefaultScopeCollectionChannelHistory map[string]auth.GrantHistory                       `json:"_default._default_channel_history,omitempty"`
```

Comment:
As discussed - let's try to get grant source into TimedSet so we're able to build a full history with source included for all cases.